nodejs/awsx/package.json

```diff
@@ -10,17 +10,23 @@
     ],
     "homepage": "https://pulumi.io",
     "repository": "https://github.com/pulumi/pulumi-awsx",
+    "scripts": {
+        "prepare": "install-peers"
+    },
     "dependencies": {
-        "@pulumi/pulumi": "^1.9.1",
-        "@pulumi/aws": "^1.0.0",
-        "@pulumi/docker": "^1.0.0",
         "@types/aws-lambda": "^8.10.23",
         "mime": "^2.0.0"
     },
+    "peerDependencies": {
+        "@pulumi/aws": "^1.0.0 || ^2.0.0",
+        "@pulumi/docker": "^1.0.0 || ^2.0.0",
+        "@pulumi/pulumi": "^1.9.1 || ^2.0.0"
+    },
```

Comment:
Why was this necessary as part of these changes?

It may well be the "right" thing to do - but it definitely could also introduce a new set of upgrade challenges (for example, many folks will now need to explicitly include `@pulumi/docker` a you had to do here in examples.

Reply:
Let me test this real quick, but I don't think we can easily take `@pulumi/docker` out. The reason being that not making it a peerDependency introduces the possibility that `awsx` will now bring in its own version of `@pulumi/pulumi` to satisfy the `@pulumi/docker` dependency. This would then break the mixed version SxS test. If I cannot repro the break, then I'll take out `@pulumi/docker`

Reply:
Looks like I was wrong about this. Reasoning through it, it's because there's no interaction between any potential `@pulumi/pulumi` that `@pulumi/docker` might bring in via ComponentResource (we never interact with `docker.Image` in `awsx`). I'll undo the `@pulumi/docker` stuff as a result.